core/store/src/trie/mod.rs

```diff
@@ -586,6 +586,130 @@ impl Trie {
         Ok(())
     }
 
+    // Prints the trie nodes starting from hash, up to max_depth depth.
+    pub fn print_recursive(&self, hash: &CryptoHash, max_depth: u32) -> String {
```

Comment:
Perhaps pass output as argument so we don’t need to collect it to String first?
```suggestion
    pub fn print_recursive(&self, out: &mut dyn std::io::Write, hash: &CryptoHash, max_depth: u32) -> String {
```

Reply:
maybe I'm missing something: but how does it help?

As the caller would still have to do:
let mut buf = String::new();
print_resursive(&buf, ...)
println!("{}", buf)

right?
(or would the caller be able to do something smarter?)

Reply:
Just pass `&mut std::io::stdout().lock()` without accumulating into a string.